plat/common/memory.c

```diff
@@ -91,6 +92,13 @@ void *ukplat_memregion_alloc(__sz size, int type, __u16 flags)
 		ostart = mrd->pbase;
 		olen   = mrd->len;
 
+		/* Check whether we are allocating from an in-image memory hole
```

Comment:
Some ELF segments are separated by holes less than page-size, due to alignments and my coalesce function would mark those holes as free memory so that nothing goes to waste. If you allocate from there say a stack or cmdl through memregion_alloc, it would have MEMRF_MAP. And you get an EEXIST, because the ELF segments are mapped there already.